draft-private-access-tokens.md

```diff
@@ -1084,11 +1049,173 @@ enc, context = SetupBaseR(enc, skI, "AccessTokenRequest")
+   generator by this blind, yielding a blinded key and blinded generator.
+1. Compute a Schnorr proof-of-knowledge demonstrating knowledge of the
+   discrete log of the blinded key with respect to the blinded generator.
+1. Serialize the blinded key, blinded generator, and proof, yielding
+   client_origin_index_key, client_origin_index_base, and client_origin_index_proof,
+   and concatenate each, yielding client_origin_index_request.
+1. Serialize the blind, yielding client_origin_index_blind.
+1. Output client_origin_index_blind, client_origin_index_request.
+
+In pseudocode, this is as follows:
+
+~~~
+// Generate blind, blinded key, blinded generator, and the proof
+blind = RandomScalar()
+blind_key = blind * CLIENT_KEY
+blind_generator = blind * Generator()
+blind_proof = SchnorrProof(CLIENT_SECRET, blind_generator, blind_key)
+
+// Serialize and produce outputs
+client_origin_index_proof = SerializeProof(blind_proof)
+client_origin_index_key = SerializeElement(blind_key)
+client_origin_index_base = SerializeElement(blind_generator)
+client_origin_index_request = concat(client_origin_index_key,
+    client_origin_index_base, client_origin_index_proof)
+client_origin_index_blind = SerializeScalar(blind)
+~~~
+
+### Mediator Behavior {#mediator-stable-mapping}
+
+Given a client key (CLIENT_KEY), client_origin_index_blind, and client_origin_index_request,
+Mediators verify the proof for correctness as follows:
+
+1. Deserialize client_origin_index_blind, yielding blind. If this fails, abort.
+1. Parse client_origin_index_request as client_origin_index_key, client_origin_index_base, and
+   client_origin_index_proof, and deserialize each to yield the blinded key, blinded generator,
+   and blind proof.
+1. Multiply CLIENT_KEY and the group generator by blind, yielding a blinded key and
+   blinded generator. If these do not match the deserialized blinded key and generator,
+   abort.
+1. Verify the blind proof against the blinded key and generator. If proof verification
+   fails, abort.
+
+In pseudocode, this is as follows:
+
+~~~
+blind = DeserializeScalar(client_origin_index_blind)
+
+// Verify the proof parameters against the client's public key
+client_origin_index_key, client_origin_index_base, client_origin_index_proof =
+    parse(client_origin_index_request)
+expected_blind_key = DeserializeElement(client_origin_index_key)
+expected_blind_generator = DeserializeElement(client_origin_index_base)
+blind_key = blind * CLIENT_KEY
+blind_generator = blind * Generator()
+if expected_blind_key != blind_key:
+    raise InvalidParameterError
+if expected_blind_generator != blind_generator:
+    raise InvalidParameterError
+
+// Verify the proof
+proof = DeserializeProof(client_origin_index_proof)
+valid = SchnorrVerify(blind_generator, blind_target, proof)
+if not valid:
+   raise InvalidProofError
+~~~
+
+Given an Issuer response client_origin_index_result, Client blind, and Client public
```

Comment:
Maybe subsection each half of the mediator protocol? Wonder if it should just be split in protocol order:

Client Behavior
Mediator Behavior (Client Key Verification)
Issuer Behavior
Mediator Behavior (Output)

Though this seems fine.